packages/terra-table/src/subcomponents/Row.jsx

```diff
@@ -203,6 +206,8 @@ function Row(props) {
             firstRowId={firstRowId}
             lastRowId={lastRowId}
             columnHighlightColor={displayedColumns[cellColumnIndex].columnHighlightColor}
+            columnSpan={columnSpan}
```

Comment:
The cell should not need the column span information.  A Cell component cannot span columns.